libcontainer/process_linux.go

```diff
@@ -203,6 +204,28 @@ func (p *setnsProcess) start() (retErr error) {
 
 	// Get the "before" value of oom kill count.
 	oom, _ := p.manager.OOMKillCount()
+	useClone3 := false
+	if cgroups.IsCgroup2UnifiedMode() && p.initProcessPid != 0 {
+		initProcCgroupFile := fmt.Sprintf("/proc/%d/cgroup", p.initProcessPid)
+		initCg, initCgErr := cgroups.ParseCgroupFile(initProcCgroupFile)
+		if initCgErr == nil {
+			if initCgPath, ok := initCg[""]; ok {
+				useClone3 = true
+				initCgDirpath := filepath.Join(fs2.UnifiedMountpoint, initCgPath)
+				fd, err := os.Open(initCgDirpath)
+				if err != nil {
+					return fmt.Errorf("error opening cgroup dir %q: %w", initCgDirpath, err)
+				}
+				defer fd.Close()
+				if p.cmd.SysProcAttr == nil {
+					p.cmd.SysProcAttr = &syscall.SysProcAttr{}
+				}
+				p.cmd.SysProcAttr.UseCgroupFD = true
+				p.cmd.SysProcAttr.CgroupFD = int(fd.Fd())
```

Comment:
man clone3 says this is available since linux 5.7. You are setting useClone3 to true, but I don't think that is detecting `CLONE_INTO_CGROUP` is supported in this kernel, right? We will need to improve the detection. Not sure about the golang wrapper, but IIRC @kolyshkin wrote that for Go. He might have tips :)

Reply:
Usually a "dummy" exec of the syscall (with some nil params or whatever) is enough to know if it's supported. I guess something similar should be possible for this clone_into_cgroup param.

Usually that is recommended because people might backport it to old kernels, and if you just check the version you might not use it when it is available.